packages/runtime/runtime-utils/src/dataStoreHelpers.ts

```diff
@@ -8,26 +8,95 @@ import {
     IFluidObject,
     IFluidRouter,
     IRequest,
+    IResponse,
 } from "@fluidframework/core-interfaces";
 import {
     IFluidDataStoreFactory,
     IFluidDataStoreRegistry,
     IProvideFluidDataStoreRegistry,
 } from "@fluidframework/runtime-definitions";
 
+interface IResponseException extends Error {
+    errorFromRequestFluidObject: true;
+    message: string;
+    code: number;
+    stack: string;
+}
+
+export function getStack() {
+    const err = new Error();
+    if (err.stack !== undefined) {
+        return err.stack;
+    }
+    try {
+        throw err;
+    } catch (err2) {
+        return (err2 as Error).stack;
+    }
+}
+
+export function exceptionToResponse(err: any): IResponse {
+    const status = 500;
+    // eslint-disable-next-line no-null/no-null
+    if (err !== null && typeof err === "object" && err.errorFromRequestFluidObject === true) {
+        const responseErr: IResponseException = err;
+        return {
+            mimeType: "text/plain",
+            status,
+            value: responseErr.message,
+            stack: responseErr.stack ?? getStack(),
+        };
+    }
+    return {
+        mimeType: "text/plain",
+        status,
+        value : `${err}`,
+        stack: getStack(),
+    };
+}
+
+export function responseToException(response: IResponse, request: IRequest) {
+    // add one more 'frame' to message
+    const message = `${response.value}\nNot found: ${request.url}`;
+    const err = new Error(message);
+    const responseErr = err as any as IResponseException;
+    responseErr.errorFromRequestFluidObject = true;
+    responseErr.message = message;
+    responseErr.code = response.status;
+    if (response.stack !== undefined) {
+        try {
+            // not clear if all browsers allow overwriting stack
+            responseErr.stack = response.stack;
```

Comment:
if this fails can we write it in a new property as loosing the original stack makes the error useless

Reply:
I'm a bit afraid to do so, as that increases the space of leaking PII (as stack will contain actual error message that may need to be striped by telemetry pipeline). So I'd rather stick with existing well defined properties, and rely on browsers that would work (Chrome works here) - we likely do not need data from all browsers, unless errors are browser specific (which should not be the case in case of request resolution). We can always change it later if we learn it's not enough.